.bumpversion.cfg

```diff
@@ -1,15 +1,18 @@
 [bumpversion]
 current_version = 1.5.0a1
-parse = (?P<major>\d+)
-	\.(?P<minor>\d+)
-	\.(?P<patch>\d+)
-	((?P<prekind>a|b|rc)
-	(?P<pre>\d+)  # pre-release version num
-	)(\.(?P<nightly>[a-z..0-9]+)
+
+# expected matches: `1.5.0`, `1.5.0a1`, `1.5.0a1.dev123457+nightly`
+# excepted failures: `1`, `1.5`, `1.5.2-a1`
+parse = (?P<major>[\d]+) # major version number
+	\.(?P<minor>[\d]+) # minor version number
+	\.(?P<patch>[\d]+) # patch version number
+	(((?P<prekind>a|b|rc) # optional pre-release type
+	?(?P<num>[\d]+?)) # optional pre-release version number
+	\.?(?P<nightly>[a-z0-9]+\+[a-z]+)? # optional nightly release indicator
```

Comment:
would need to give this some thought but I think we want `1.5.0a1` to be valid but `1.5.0a` to be invalid right?

Reply:
the `+` in `?(?P<num>[\d]+?))` should handle that.

Reply:
I missed the parenthesis

Reply:
@colin-rogers-dbt `1.5.0a` is not valid because of the `?` in the middle.  it means what come before is optional, but if it's filled, what's after is required.
`((?P<prekind>a|b|rc)?(?P<num>[\d]+?))`